### DISCUSSION TITLE
Shuffle upgrade app list

### DISCUSSION THREAD

**Discussion by Romanitho** [Romanitho/Winget-AutoUpdate#137] [Ideas]
I encountered an issue with a package. In my case, Inkscape remains stuck on upgrade and blocks the WAU queue. Each time WAU runs, it gets blocked on Inkscape upgrade... All the other updates cannot be upgraded then.

To workaround this, we could configure WAU to shuffle the upgrade list to update apps randomly each time it is executed. Quit easy to implement.

Or, if someone has a better idea to manage this kind of problem, I take.

**Comment by Romanitho** [comment]
@KnifMelti what do you think ? 😅

**Comment by KnifMelti** [comment]
One could argue that it's Inscapes fault, but I think a shuffle is an excellent idea!

**Reply by KnifMelti** [reply]
...or/and, you could introduce some sort of upgrade timer...

**Reply by KnifMelti** [reply]
...if it's not related to a Pending Reboot.
Kill the winget.exe process...

**Reply by Romanitho** [reply]
Yes, timer could be something interesting, but it really depend on the size of the app and time needed to install/upgrade it.
Shuffle is quite easy to implement. We can still add a timeout (60 min per app? ^^)
